examples/nsl_pk/DY.Example.NSL.Protocol.Stateful.Proof.fst

```diff
@@ -54,15 +54,15 @@ let event_predicate_nsl: event_predicate nsl_event =
     match e with
     | Initiate1 alice bob n_a -> (
       prin == alice /\
-      get_label n_a == join (principal_label alice) (principal_label bob) /\
-      0 < DY.Core.Trace.Type.length tr /\
-      rand_generated_at tr (DY.Core.Trace.Type.length tr - 1) n_a
+      is_secret (join (principal_label alice) (principal_label bob)) tr n_a /\
+      0 < DY.Core.Trace.Base.length tr /\
```

Comment:
I know this isn't the point of the changes, but this makes me realize that we might want this wrapped up as a predicate (something like `trace_non_empty`?) just to make things more explicit.

Reply:
The change here is twofold:
- `length` moved from `DY.Core.Trace.Type` to `DY.Core.Trace.Base`, so it's changed in `0 < ...` and `rand_generated_at`
- `get_label ... == ...` is changed to `is_secret ...` to also requires `bytes_invariant` on `n_a`. This is useful to have `bytes_well_formed tr n_a` and use the theorem on `get_label` when the trace grows.

I agree that things could look nicer here but this should belong to a separate PR. I would lean toward keeping `0 < ...` or `1 <= ...`, but renaming `DY.Core.Trace.Base.length` to `trace_length` so that we don't need to specify the namespace everytime. It would look like `0 < trace_length tr` which is not too verbose.

Reply:
I prefer the `non_empty` version for the trace. Btw, why is it not just `Snoc? tr`. It should be the same.

Reply:
Moving the discussion in #61.